### DISCUSSION TITLE
Is there any way to login by facebook or twitter using apis

### DISCUSSION THREAD

**Discussion by alihassan161820** [directus/directus#10058] [Archive]
I'm trying to access sso api login after adding sso env variables 
Post :  /auth/login/facebook and always return with route not found despite of there are social login buttons in login page in directus dashboard, so it there is anything i missed ?

### To Reproduce

setting env variables 

```
AUTH_PROVIDERS="google,facebook"
AUTH_FACEBOOK_DRIVER="oauth2"
AUTH_FACEBOOK_SCOPE="id"
AUTH_FACEBOOK_CLIENT_ID="string:324---------6"
AUTH_FACEBOOK_CLIENT_SECRET="string:c6---------------2"
AUTH_FACEBOOK_AUTHORIZE_URL="https://www.facebook.com/dialog/oauth"
AUTH_FACEBOOK_ACCESS_URL="https://graph.facebook.com/oauth/access_token"
AUTH_FACEBOOK_PROFILE_URL="https://graph.facebook.com/me"
```

and trying to post sso login api  
auth/login/facebook

**Comment by rijkvanzanten** [comment]
Heya! 

You won't send any data to the /auth/login/facebook endpoint. Instead, the browser should open that link, so that the user is redirect to facebook.com itself. Try adding a `<a href="https://example.com/auth/login/facebook?redirect=https://my-app.com">Login with Facebook</a>`. Where example.com is the url of your directus instance, and my-app.com where you're redirecting from 👍🏻

**Reply by alihassan161820** [reply]
@rijkvanzanten 
Thanks for your reply
But i need to use this endpoint in mobile app

**Reply by joselcvarela** [reply]
Hey @alihassan161820 
You need to open a webview then. It is mandatory to do so because your users will need to authenticate into twitter or facebook and for that you should use an webview. If you are using some third party framework you need to figure out how you can open this link `https://example.com/auth/login/facebook?redirect=https://my-app.com` (replace strings as rijk mentioned)

**Reply by alihassan161820** [reply]
is directus support apple login ? 
if not is there any alternative solution to execute it ?
@joselcvarela @rijkvanzanten

**Reply by rijkvanzanten** [reply]
Directus supports any OAuth2 / OpenID / LDAP compatible service

**Reply by gkielwasser** [reply]
> Hey @alihassan161820 You need to open a webview then. It is mandatory to do so because your users will need to authenticate into twitter or facebook and for that you should use an webview. If you are using some third party framework you need to figure out how you can open this link `https://example.com/auth/login/facebook?redirect=https://my-app.com` (replace strings as rijk mentioned)

It seems not possible anymore to login Facebook from a webview. Probably because of the following
See also https://developers.facebook.com/docs/facebook-login/android/deprecating-webviews

What would be an alternative ?

**Comment by AKhateeb** [comment]
I have the same issue and all attempts failed to get it working, I'm using only "Login with Gmail" with my Directus as integrating Facebook has failed
